### ISSUE TITLE
new rule - property type mismatch

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ty#3633]
```py
class Foo:
    @property
    def value(self) -> int: ...

    @value.setter
    def value(self, new_value: str): ... # error
```

while this may be intentional, i consider it more of a lint rule to avoid overusing properties and creating confusing APIs.

pyright has a rule for this: `reportPropertyTypeMismatch`

**Comment by AlexWaygood** [comment]
This would be an opinionated lint rule that isn't part of the typing spec and isn't really about type safety. I can see that some might find it useful, but it would also have false positives, so I don't think we'd want to ever enable it by default (I think there's lots of idiomatic code that would be flagged by this). It's possible we might add something like it at some point when we look at building a type-aware linter on top of ty, but I'm not sure it fits on our current roadmap.
